src/Interactive/EditorFeatures/Core/Extensibility/Interactive/NuGetPackageResolverImpl.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+extern alias WORKSPACES;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Roslyn.Utilities;
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.CodeAnalysis.Editor.Interactive
+{
+    internal sealed class NuGetPackageResolverImpl : WORKSPACES::Microsoft.CodeAnalysis.Scripting.NuGetPackageResolver
+    {
+        private readonly string _packagesDirectory;
+
+        internal NuGetPackageResolverImpl(string packagesDirectory)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(packagesDirectory));
+            Debug.Assert(PathUtilities.IsAbsolute(packagesDirectory));
+
+            _packagesDirectory = packagesDirectory;
+        }
+
+        internal override ImmutableArray<string> ResolveNuGetPackage(string reference)
```

Comment:
We might be able to do it for #r, but it can't be done for #load since the loaded script may contain #load. We could resolve all #loads within one file.